truffleruby/src/main/java/org/truffleruby/core/CoreLibrary.java

```diff
@@ -1099,6 +1099,10 @@ public static boolean fitsIntoInteger(long value) {
         return ((int) value) == value;
     }
 
+    public static boolean fitsIntoUnsignedInteger(long value) {
+        return value == (value & 0xffffffffl) || value < 0 && value >= Integer.MIN_VALUE;
```

Comment:
So, I would hope nothing after the `||` is ever relevant, but the rule in MRI appears to be as follows
1. Anything which can fir into a signed int (positive or negative) can be turned into an unsigned int
2. Anything integer which can be represented in 32 bits can be turned into an unsigned int.

So a small negative long can be cast to an unsigned int because it can be simply truncated to a signed int without loss of meaning and then cast to be unsigned. This function is not normally passed small negative longs in our implementation, but I put the check in there so it would give the correct answer even if we slipped up elsewhere.